src/dash-table/reconcile/number.ts

```diff
@@ -1,33 +1,17 @@
-import { INumberColumn } from 'dash-table/components/Table/props';
-import isNully from './isNully';
+import isNumeric from 'fast-isnumeric';
 
-export function coerce(value: any, options?: INumberColumn) {
-    const validation = validate(value, options);
-    if (validation.success) {
-        return validation;
-    }
+import { INumberColumn } from 'dash-table/components/Table/props';
+import { reconcileNull } from './null';
+import { IReconciliation } from '.';
 
-    switch (typeof value) {
-        case 'string':
-            return { success: Boolean(/(-)?\d+([.]\d+)?/.test(value)), value: parseFloat(value) };
-        default:
-            return { success: false, value };
-    }
+export function coerce(value: any, options: INumberColumn | undefined): IReconciliation {
+    return isNumeric(value) ?
+        { success: true, value: +value } :
+        reconcileNull(value, options);
 }
 
-export function validate(value: any, options?: INumberColumn) {
-    const allowNull = Boolean(options && options.validation && options.validation.allow_null);
-
-    const nully = isNully(value);
-    if (nully && allowNull) {
-        return { success: true, value: null };
-    }
-
-    value = nully ? null : value;
-
-    if (typeof value === 'number') {
-        return { success: !nully, value };
-    } else {
-        return { success: false, value };
-    }
+export function validate(value: any, options: INumberColumn | undefined): IReconciliation {
+    return typeof value === 'number' && !isNaN(value) ?
```

Comment:
`isFinite` will filter out `+/-Infinity` in addition to `NaN`

Reply:
Checked that `+/-Infinity` also resolves to `null` -- will add them to the nully cases

Reply:
Ah right - that'll keep us pretty safe. If anyone ever finds a way around that we can address it then 😎